lib/parse-client-options.js

```diff
@@ -38,18 +55,11 @@ function parseOptions (userOptions) {
     }
   }
 
-  clientDefaults.baseUrl = `${options.protocol}://${options.host}`
-
-  if (options.port) {
-    clientDefaults.baseUrl += `:${options.port}`
-  }
-
-  // Check if a prefix is passed in the options and strip any leading or trailing slashes from it.
-  if (options.pathPrefix) {
-    clientDefaults.baseUrl += '/' + options.pathPrefix.replace(/(^[/]+|[/]+$)/g, '')
+  if (options.baseUrl) {
+    clientDefaults.baseUrl = `${options.baseUrl}`
```

Comment:
no need to wrap it in `${}`, just `clientDefaults.baseUrl = options.baseUrl` should work.

The previous implementation you had before looked good, why did you remove it? Now we won’t support the `protocol`, `host`, `port` and `pathPrefix` argument any longer. But we still want to support them so apps don’t break, they will only log the deprecation warning.

Reply:
@gr2m I must have done that by mistake. I was playing around with some things. 😅 
Also, i'm having some trouble to pass some tests given in `test/integration/agent-proxy` .
I've tried but i don't think i found the source of that error. Could you help me solving it?  😄